jpl.pl

```diff
@@ -930,11 +946,12 @@
 % NB this does not yet handle shadowed fields correctly.
 
 jpl_set_static(Type, ClassObj, Fname, V) :-
+    Pred=jpl_set/3,
     (   atom(Fname)                     % the usual case
     ->  true
     ;   var(Fname)
-    ->  throw(error(instantiation_error,context(jpl_set/3,'when 1st arg denotes a class, 2nd arg must be bound to the name of a public, static, non-final field')))
-    ;   throw(error(type_error(field_name,Fname),context(jpl_set/3,'when 1st arg denotes a class, 2nd arg must be the name of a public, static, non-final field')))
+    ->  throw_instantiation_error(Pred,pl_set/3,ep01)
+    ;   throw_type_error(field_name,Fname,ep02)
```

Comment:
@dtonhofer , I think here is the problem! It is using `throw_type_error/3` that does not exist because there is only `throw_type_error/4`

Basically it is missing `Pred` as first argument...

Reply:
Good find. Will fix.

Reply:
Ok, fixed this one. How are such problems detected usually before unit testing? When writing Java code, the compiler makes you feel as if you were in a valley where you are safely kept, but Prolog is like wandering a mountain ridge. Shurely there must be a linter, similar to the excellent shellcheck for bash scripts.

Reply:
> How are such problems detected usually before unit testing?

The built-in PceEmacs reports a lot of issues as red colored code.  check/0 also does a lot of useful stuff.  gxref/0 can be used to find dead (unreachable) code.

Reply:
I need to level up

Reply:
I'll wait for a message "go ahead" ...

Reply:
I would say "go ahead".